test/integration/test_http.py

```diff
@@ -73,5 +73,196 @@ def test_basicTxn(self):
+        result = self.sendReq("/api/Write", ID, schema="test_schema1")
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Write with bad schemaVersion, should fail
+        result = self.sendReq("/api/Write", ID, ver=2)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Write with bad partition value data type, should fail
+        result = self.sendReq("/api/Write", ID, partition=1)
+        self.assertEqual(result["code"], 500)
+
+        # Write with bad range data type, should fail
+        result = self.sendReq("/api/Write", ID, range_key=1)
+        self.assertEqual(result["code"], 500)
+
+        # Write with bad data field data type, should fail
+        result = self.sendReq("/api/Write", ID, data=1)
+        self.assertEqual(result["code"], 500)
+
+        # Try read data written by failed requests, should fail
+        result = self.sendReq("/api/Read", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Do a valid write, should succeed
+        result = self.sendReq("/api/Write", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 201)
+
+        # Read with bad Txn ID, should fail
+        result = self.sendReq("/api/Read", ID+1)
+        result = result[0]
+        self.assertEqual(result["status"]["code"], 400)
+
+        # Read with bad collectionName, should fail
+        result = self.sendReq("/api/Read", ID, coll="HTTPClient1")
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Read with bad schemaName, should fail
+        result = self.sendReq("/api/Read", ID, schema="test_schema1")
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Read with bad partition data type, should fail
+        result = self.sendReq("/api/Read", ID, partition=3)
+        self.assertEqual(result["code"], 500)
+
+        # Read with bad range data type, should fail
+        result = self.sendReq("/api/Read", ID, range_key=3)
+        self.assertEqual(result["code"], 500)
+
+        # Do a valid read, should succeed
+        result = self.sendReq("/api/Read", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 200)
+
+        # Commit, with bad Txn ID, should fail
+        result = self.sendReq("/api/EndTxn", ID+1)
+        result = result[0]
+        self.assertEqual(result["status"]["code"], 400)
+
+        # Commit, should succeed
+        result = self.sendReq("/api/EndTxn", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 200)
+
+        # Commit again, should fail
+        result = self.sendReq("/api/EndTxn", ID)
+        result = result[0]
+        self.assertEqual(result["status"]["code"], 400)
+
+    # Test read write conflict between two transactions
+    def test_read_write_txn(self):
```

Comment:
To get this test to work you need a sequence like:

- Begin Txn1
- Begin Txn2
- Read key with Txn2 (404 expected)
- Write key with Txn1 (conflict error code expected)

What you have is trying to read a record with an old timestamp, which is not an error for us because of MVCC. I can explain more if you want.

Reply:
Fixed. Populated initial value so that step 3 succeeds.